roles/lib_utils/library/docker_creds.py

```diff
@@ -125,7 +131,23 @@ def load_config_file(module, dest):
         return {}
 
 
-def update_config(docker_config, registry, username, password):
+def validate_registry_login(module, registry, encoded_auth):
+    '''Attempt to use credentials to log into registry'''
+    url = 'https://' + registry + '/v2/'
```

Comment:
@sdodson Apparently this might not be a perfect solution for checking if a registry other than ours is working.  Any thoughts?

Reply:
@bparees Mike is extending the existing module which writes out a docker config.json to add a validation that the credentials are accepted. Do you know if the method here will be robust at least for docker hub and Red Hat's registries?

We use this because of a few reasons, mainly because we do this early in installation process before the docker daemon may be running and because `docker login` provides no differentiation between the scenario where you're already logged in and a successful new login. Which is required for us to know when to trigger other actions such as copying the file to /var/lib/origin/.docker/config.json and restarting the kubelet.

Reply:
basic auth (using the username and then the token as password) should work fine with the new terms based registry, that's how openshift intends to interact with it.

for a more detailed confirmation of the exact flow you're doing here, we'd need to talk to the TBR team.  I don't know their github ids but you'd want josh cain or patrick easters, most likely.